.yarnrc

```diff
@@ -0,0 +1,5 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+yarn-path ".yarn/releases/yarn-1.22.11.cjs"
```

Comment:
Could you please explain me what is the reason to have it here?

Reply:
https://classic.yarnpkg.com/lang/en/docs/yarnrc/
* https://yarnpkg.com/configuration/yarnrc

The point of having is to developers and tools have the same environment.

In this case, seems the project is configured for yarn 1, is that true?

Reply:
Yes, it is version 1. It looks like putting an .yarnrpc and moreover the whole code of yarn in the repo is a bit excessive. Why not just add `packageManager` param to package.json?

Reply:
that question is for you and @wainola

Reply:
What do you mean by that? @BeroBurny

Reply:
@enemycnt with packageManager it will work as well, however from the package.json dock it's mentioned that the feature is still experimental, so yarnrc is probably more reliable.

> Defines which package manager is expected to be used when working on the current project. This field is currently experimental and needs to be opted-in; see https://nodejs.org/api/corepack.html

Reply:
> What do you mean by that? @BeroBurny

Nobody set `packageManager` and saying that `.yanrrc` excessive,asumbing by that is there reason why is not set?

Reply:
@BeroBurny But what is the reason to have the whole code of yarn in `.yarn/releases` instead of just mentioning the version? Maybe you can explain to me some specific case for that?
@Lykhoyda Yes, I agree with you, but maybe having version in engines could be better than having the whole code?
like this
````
{
  "engines": {
    "npm": "~1.0.20"
  }
}

Reply:
@enemycnt in the engines you set only the supported range of the package manager, but installing part is on the user. You could have globally different versions installed. The command generates the .yarn folder:
```bash
yarn policies set-version 
```

Advantages of this approach you can find here: https://classic.yarnpkg.com/en/docs/cli/policies/

Also, I had an issue installing the packages using yarn, so strictly setting the version solved the problem.

Reply:
Using `engines` this way can bring pain with a compatiblities.

Situation A (using your example)
```json
{
  "engines": {
    "npm": "~1.0.20"
  }
}
```
Anyone with different version of like you `Bob` have npm version `3.0.0` will be "forced" to install version `1.0.x` and when `Bob` is back to other project he will need to install back version `3.0.0`

Situation B (from project)
```json
{
  "engines": {
    "node": ">=12",
    "yarn": ">=1"
  },
}
```
`Bob` with yarn version 2 or above can have troubles while contributing on a project, with a lot of questions or roadblocks
Take in consideration that he can even have outdated version.

Situation C (using `packageManager`)
```json
{
  "packageManager": "1.22.19"
}
```

Is good approach `Bob` dose not need to manually install anything (think he youst need to enable `corepack ` by using `corepack eneable` in terminal) but that is for node version 16 and above
But there is gotcha.

In this project node engine setting are like this 
```json
{
  "engines": {
    "node": ">=12",
    "yarn": ">=1"
  },
}
```
That means if `Bob` for example uses node version 14 he would have `corepack` as default and to be able to use it he need to install dependency (for example `npm i -g corepack`)
It is not painful but need to be visibly documented to avoid questions why is not working

Reply:
According to [npm docs](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#engines) 

> Unless the user has set the [engine-strict config](https://docs.npmjs.com/cli/v9/using-npm/config#engine-strict) flag, this field is advisory only and will only produce warnings when your package is installed as a dependency.

So in case of other package manager or incompatible version only warning will appear. We are not using egines-strict flag here. We just give a hint, not forcing. 

In the next PR by @Lykhoyda , the [version 16 of node is added to engines.](https://github.com/sygmaprotocol/sygma-ui/blob/e68cdea90cc65f79cc7c909725d17fc712407d52/package.json#L11) I welcome this. I think, even better to use 18 as current lts release of node.
So it makes sense to put  `packageManager` here, to have accurate version of yarn here and get read of `.yarnrc` and `.yarn/releases`

Reply:
@enemycnt it's precisely the problem that you get only a warning and must do additional steps to install the correct version. In the case of yarnrc file, we tackle the problem with "plug n play" approach, the user shouldn't know about all hidden knowledge about yarn. Moreover, we know that with the version of the packages during installation we will receive an error if the correct yarn version will not be used. Using "packageManager" will not resolve the cases that Bero mentioned above.

The requirement for the Node js 16 version is updated in the next PR, but you still have to enable corepack which is an experimental feature. Also, I would postpone Node Js update to version 18, until we update the project dependencies to the latest version as migration to version 18 normally has breaking changes.

Reply:
Thanks, @Lykhoyda 
If it is gonna help at least yarn users, that is fine.